engines/dfc_provider/app/services/affiliate_sales_data_builder.rb

```diff
@@ -0,0 +1,171 @@
+      sales_data.map { |sale| build_order(sale) }
+    end
+
+    # def build_sale_sessions
+    #   sales_data.map { |sale| build_sale_session(sale) }
+    # end
+
+    private
+
+    def build_enterprise(id)
+      DataFoodConsortium::Connector::Enterprise.new(urls.enterprise_url(id))
+    end
+
+    def build_address(sale)
+      DataFoodConsortium::Connector::Address.new(
+        urls.address_url(sale.producer_id),
+        postalCode: sale.producer_postcode,
+        country: nil,
+        latitude: nil,
+        longitude: nil,
+        region: nil,
+        street: nil,
+        city: sale.producer_city
+      )
+    end
+
+    def build_producer(sale)
+      DataFoodConsortium::Connector::Enterprise.new(
+        urls.enterprise_url(sale.producer_id),
+        logo: nil,
+        name: nil,
+        description: nil,
+        vatNumber: nil,
+        suppliedProducts: build_supplied_product(sale),
+        localizations: build_address(sale)
+      )
+    end
+
+    def build_supplied_product(sale)
+      DataFoodConsortium::Connector::SuppliedProduct.new(
+        urls.enterprise_supplied_product_url(sale.producer_id, sale.product_id),
+        name: sale.product_name,
+        description: nil,
+        alcoholPercentage: nil,
+        lifetime: nil,
+        usageOrStorageConditions: nil,
+        totalTheoreticalStock: nil,
+        catalogItems: build_catalogue_item(sale),
+        quantity: build_quantity(sale)
+      )
+    end
+
+    def build_catalogue_item(sale)
+      DataFoodConsortium::Connector::CatalogItem.new(
+        urls.enterprise_catalog_item_url(sale.producer_id, sale.product_id),
+        stockLimitation: nil,
+        sku: nil,
+        offers: build_offer(sale)
+      )
+    end
+
+    def build_offer(sale)
+      DataFoodConsortium::Connector::Offer.new(
+        urls.enterprise_offer_url(sale.producer_id, sale.product_id),
+        stockLimitation: nil,
+      )
+    end
+
+    def build_order_line(sale)
+      DataFoodConsortium::Connector::OrderLine.new(
+        urls.enterprise_order_order_line_url(sale.producer_id, sale.order_id, sale.line_item_id),
+        description: nil,
+        offer: build_offer(sale),
+        order: build_order(sale),
+        quantity: build_quantity(sale),
+        price: build_price(sale)
+      )
+    end
+
+    def build_order(sale)
+      DataFoodConsortium::Connector::Order.new(
+        urls.enterprise_order_url(sale.producer_id, sale.order_id),
+        number: nil,
+        date: sale.order_date.strftime("%Y-%m-%d")
+        # saleSession: build_sale_session(sale)
+      )
+    end
+
+    def build_sale_session(sale)
+      DataFoodConsortium::Connector::SaleSession.new(
+        urls.enterprise_sale_session_url(sale.producer_id, sale.line_item_id),
+        beginDate: nil,
+        endDate: nil,
+        quantity: nil
+      )
+    end
+
+    def build_quantity(sale)
+      DataFoodConsortium::Connector::Quantity.new(
+        unit: sale.unit_presentation,
```

Comment:
I think this should be a constant. Have a look at this code:

https://github.com/openfoodfoundation/openfoodnetwork/blob/d60d29b6853dc9726027ce310c0317e91afd7fa1/engines/dfc_provider/app/services/quantitative_value_builder.rb#L19-L28

I didn't actually know that there was a `Quantity` class as well. I don't know what's used in which case. @RaggedStaff ?